pkg/scheduler/api/types.go

```diff
@@ -53,10 +59,40 @@ const (
 	Unknown
 )
 
+// JobReadiness type of job readiness
+type JobReadiness int
+
+const (
+	// Ready : a job is Ready if the number of tasks in Allocated state
+	// exceeds the job's minimum task number requirement. In other words,
+	// #(Allocated Tasks) >= Job.MinAvailable
+	// A Ready job can be dispatch to a node right away.
+	Ready JobReadiness = 1 << iota
```

Comment:
we dependent on plugin to decide whether a job is ready, refer to `plugin/gang` & `JobReady` callback.

Reply:
fixed

Reply:
how did we fix that? I mean we should not have such kind of info in framework, framework is hard to decide whether a 'job' is ready, it all dependent on plugin to make decision. As we're going to support different type of workload.